ToolBox/IEX/IEXDataQueueHandler.cs

```diff
@@ -418,95 +384,124 @@ private IEnumerable<Slice> ProcessHistoryRequests(Data.HistoryRequest request)
-                            var mins = TimeSpan.ParseExact(item["minute"].Value<string>(), "hh\\:mm", CultureInfo.InvariantCulture);
-                            date += mins;
-                        }
-                        else
-                        {
-                            date = Parse.DateTime(item["date"].Value<string>());
-                        }
+            var responses = Task.WhenAll(requests).Result;
 
-                        if (date.Date < start.Date || date.Date > end.Date)
-                        {
-                            continue;
-                        }
+            foreach (var response in responses)
+            {
+                var parsedResponse = JArray.Parse(response);
 
-                        Interlocked.Increment(ref _dataPointCount);
+                // Parse
+                foreach (var item in parsedResponse.Children())
+                {
+                    DateTime date;
+                    TimeSpan period;
+                    if (item["minute"] != null)
+                    {
+                        date = DateTime.ParseExact(item["date"].Value<string>(), "yyyy-MM-dd", CultureInfo.InvariantCulture);
+                        var minutes = TimeSpan.ParseExact(item["minute"].Value<string>(), "hh\\:mm", CultureInfo.InvariantCulture);
+                        date += minutes;
+                        period = TimeSpan.FromMinutes(1);
+                    }
+                    else
+                    {
+                        date = Parse.DateTime(item["date"].Value<string>());
+                        period = TimeSpan.FromDays(1);
+                    }
 
-                        if (item["open"].Type == JTokenType.Null)
-                        {
-                            continue;
-                        }
-                        var open = item["open"].Value<decimal>();
-                        var high = item["high"].Value<decimal>();
-                        var low = item["low"].Value<decimal>();
-                        var close = item["close"].Value<decimal>();
-                        var volume = item["volume"].Value<int>();
+                    if (date < start || date > end)
+                    {
+                        continue;
+                    }
 
-                        var tradeBar = new TradeBar(date, request.Symbol, open, high, low, close, volume);
+                    Interlocked.Increment(ref _dataPointCount);
 
-                        yield return new Slice(tradeBar.EndTime, new[] {tradeBar});
+                    if (item["open"].Type == JTokenType.Null)
+                    {
+                        continue;
                     }
+                    var open = item["open"].Value<decimal>();
+                    var high = item["high"].Value<decimal>();
+                    var low = item["low"].Value<decimal>();
+                    var close = item["close"].Value<decimal>();
+                    var volume = item["volume"].Value<int>();
```

Comment:
It seems for minute data they send the data in different format:
https://sandbox.iexapis.com/stable/stock/twtr/chart/date/20201020?token=Tsk_f255cad505354b5db620193228160965

But I am changing so it can choose between adjusted/raw at least for daily bars, - for daily they come in a format as described in docs.